### ISSUE TITLE
bug: pb_encode does not encode tag for extension field that is all zeros

### ISSUE THREAD

**Issue by lukaszgemborowski** [nanopb/nanopb#242]
edited: shorter test case

example will be the best to visualize the issue (one of my gtest test cases):
```cpp
TEST(nanopb, BUG_encode_decode_extension)
{
        uint8_t buffer[256];
        IntegerContainer source_field = { 0 };
        Extendable source = { 0 };

        pb_extension_t source_ext = { 0 };
        source_ext.type = &field_a;
        source_ext.dest = &source_field; // all zeroes - this is important
        source_ext.next = nullptr;
        source_ext.found = false;

        source.extensions = &source_ext;

        auto ostream = pb_ostream_from_buffer(buffer, sizeof(buffer));
        ASSERT_TRUE(pb_encode(&ostream, Extendable_fields, &source));

        EXPECT_NE(0, ostream.bytes_written);
}
```
result of such test case is:
```
nanopb_basic_tests.cc:120: Failure
Expected: (0) != (ostream.bytes_written), actual: 0 vs 0
```
My test.proto file is:
```
syntax = "proto2";
import "nanopb.proto";

message IntegerContainer {
        required int32 a = 1;
        optional int32 b = 2;
}

message FloatContainer {
        required float c = 1;
}

message Extendable {
        extensions 10 to 100;
}

extend Extendable {
        optional IntegerContainer field_a = 11;
        optional FloatContainer field_b = 12;
}
```
edit: to be precise, when I add: `source_field.a = 1;` in my test case it pass, meaning that bytes_written != 0. This is because of check in pb_check_proto3_default_value (else case)

**Comment by PetteriAimonen** [comment]
Hmm yeah, seems that the proto3 checks are mistakenly getting applied for extension fields also in proto 2 mode.

**Comment by lukaszgemborowski** [comment]
and honestly I do not see any easy fix for that now. Maybe trying to code some additional information in pb_field_t in extension->type->arg; so you can make a proper decision in encode_basic_field(). But I do not know nanopb implementation in many details.

**Comment by PetteriAimonen** [comment]
Seems that it is better to change how proto3 support is implemented, as I now realized that it also has the bug that it is not invoked for pointer/callback fields is proto3 files. But I'll have to think about it a bit.

**Comment by lukaszgemborowski** [comment]
Take your time. I won't came up with any solution rather soon as my knowledge of the code base is like 2 hours of reading pb_encode.c. But if you need some help with implementation or testing let me know. I do have some spare time sometimes. ;-)

**Comment by PetteriAimonen** [comment]
Should be fixed now; found a pretty simple (if not very elegant) fix.

**Comment by lukaszgemborowski** [comment]
Working correctly now. Thanks :-)
```
[ RUN      ] nanopb.BUG_encode_decode_extension
[       OK ] nanopb.BUG_encode_decode_extension (0 ms)
```

**Comment by PetteriAimonen** [comment]
Fix released in nanopb-0.3.8.
